jwql/utils/utils.py

```diff
@@ -43,6 +43,109 @@
 
 __location__ = os.path.realpath(os.path.join(os.getcwd(), os.path.dirname(__file__)))
 
+ON_GITHUB_ACTIONS = '/home/runner' in os.path.expanduser('~') or '/Users/runner' in os.path.expanduser('~')
+
+
+def _validate_config(config_file_dict):
+    """Check that the config.json file contains all the needed entries with
+    expected data types
+
+    Parameters
+    ----------
+    config_file_dict : dict
+        The configuration JSON file loaded as a dictionary
+
+    Notes
+    -----
+    See here for more information on JSON schemas:
+        https://json-schema.org/learn/getting-started-step-by-step.html
+    """
+    # Define the schema for config.json
+    schema = {
+        "type": "object",  # Must be a JSON object
+        "properties": {  # List all the possible entries and their types
+            "connection_string": {"type": "string"},
+            "database": {
+                "type": "object",
+                "properties": {
+                    "engine": {"type": "string"},
+                    "name": {"type": "string"},
+                    "user": {"type": "string"},
+                    "password": {"type": "string"},
+                    "host": {"type": "string"},
+                    "port": {"type": "string"}
+                    },
+                    "required": ['engine', 'name', 'user', 'password', 'host', 'port']
+                 },
```

Comment:
It looks like there are a few little things that pep8speaks already mentioned, such as the over-indent of this bracket, but I didn't find any additional formatting issues.